### ISSUE TITLE
remake src/Mllib/Classification/KNN.hs

### ISSUE THREAD

**Issue by vsha96** [vsha96/mllib#7]
We need to remake the [KNN](https://github.com/vsha96/mllib/blob/main/src/Mllib/Classification/KNN.hs) model according to the **basic principles** (see more in [contributing guidelines](https://github.com/vsha96/mllib/blob/main/CONTRIBUTING.md))

**Comment by s1m0000n** [comment]
Would like to start working on this. So, I basically need to make sure everything follows these guidelines https://kowainik.github.io/posts/2019-02-06-style-guide? Anything else or that's it?

**Comment by vsha96** [comment]
@s1m0000n yes, try to follow the code style in the [guidelines (basic principles)](https://github.com/vsha96/mllib/blob/main/HOW_TO_WRITE_MODEL.md) and do remake of KNN
It will be great if you close TODOs that related to metrics from [Metric.hs](https://github.com/vsha96/mllib/blob/6a6f2bc51d96c23051e35c4f7e0c9183f2c8a5c5/src/Mllib/Metrics.hs), without weights, because it may require much more changes
So you need to add the ability to select a metric , the default metric should be Euclidean

I expect that the data type `KNeighborsParams` should appear (like in [KMeans](https://github.com/vsha96/mllib/blob/6a6f2bc51d96c23051e35c4f7e0c9183f2c8a5c5/src/Mllib/Cluster/KMeans.hs#L58)) with the default parameters in the block of main functions
Therefore, the fitting function for the KNN model might be called like this

Old version:
```haskell
fitKNeighbors 3 {-number of neighs-} x y
```
New version:
```haskell
fitKNeighbors KNeighborsParams{n=3 {-number of neighs-}} x y
```
```haskell
-- | example with ability to change metric
import Mllib.Metrics (chebyshevDistance)
...
fitKNeighbors KNeighborsParams{n=5, metric=chebyshevDistance} x y
```

Feel free to change names of funcs and constants etc if you think it would be better

**Comment by s1m0000n** [comment]
You can assign me back as I'm returning to work & starting with this task again =)
